modules/vm/main.tf

```diff
@@ -131,3 +132,30 @@ resource "azurerm_linux_virtual_machine_scale_set" "graphdb" {
 
   tags = var.tags
 }
+
+resource "azurerm_role_definition" "backup_role" {
+  name        = "ReadOnlyVMSSStorageRole"
+  scope       = data.azurerm_resource_group.graphdb.id
+  description = "This is a custom role created via Terraform required for creating backups in GraphDB"
+
+  permissions {
+    actions = [
+      "Microsoft.Compute/virtualMachineScaleSets/read",
+      "Microsoft.Storage/storageAccounts/read"
+    ]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    data.azurerm_resource_group.graphdb.id
+  ]
+  depends_on = [data.azurerm_resource_group.graphdb]
+}
+
+resource "azurerm_role_assignment" "rg-reader-role" {
+  principal_id         = data.azurerm_user_assigned_identity.graphdb-instances.principal_id
+  scope                = data.azurerm_resource_group.graphdb.id
+  role_definition_name = "ReadOnlyVMSSStorageRole"
+  depends_on           = [azurerm_role_definition.backup_role, data.azurerm_resource_group.graphdb]
```

Comment:
I think that this whole `depends_on` is not needed.

Reply:
Several times I've hit the role assignment trying to assign a missing role, IMHO `azurerm_role_definition.backup_role` is required